pkg/util/util.go

```diff
@@ -63,7 +72,9 @@ func Hasher() func(string) (string, error) {
 				return "", err
 			}
 			defer f.Close()
-			if _, err := io.Copy(h, f); err != nil {
+			buf := pool.Get().(*[]byte)
```

Comment:
I am little bit confused with what this does. Basically, you are creating an entry for a new location?
Will this `sync poll` be ever to read?